crates/ball_filter/src/hypothesis.rs

```diff
@@ -28,20 +28,34 @@ pub enum BallMode {
 
 #[derive(Clone, Debug, Serialize, Deserialize, PathSerialize, PathDeserialize, PathIntrospect)]
 pub struct BallHypothesis {
+    identifier: u64,
     pub mode: BallMode,
     pub last_seen: SystemTime,
     pub validity: f32,
 }
 
 impl BallHypothesis {
-    pub fn new(hypothesis: MultivariateNormalDistribution<4>, last_seen: SystemTime) -> Self {
+    pub fn new(
+        hypothesis: MultivariateNormalDistribution<4>,
+        identifier: u64,
+        last_seen: SystemTime,
+    ) -> Self {
         Self {
             mode: BallMode::Moving(hypothesis),
             last_seen,
+            identifier,
             validity: 1.0,
         }
     }
 
+    pub fn identifier(&self) -> u64 {
+        self.identifier
+    }
+
+    pub fn set_identifier(&mut self, identifier: u64) {
+        self.identifier = identifier;
+    }
+
```

Comment:
if you have public getter and setter, why not make the member public in the first place?